lib/astutils.cpp

```diff
@@ -3142,11 +3142,14 @@ ExprUsage getExprUsage(const Token* tok, int indirect, const Settings* settings)
             return getExprUsage(tok->astParent()->astParent(), indirect, settings);
     }
     if (indirect == 0) {
-        if (Token::Match(tok->astParent(), "%cop%|%assign%|++|--") && !Token::simpleMatch(tok->astParent(), "=") &&
+        if (Token::Match(tok->astParent(), "%cop%|%assign%|++|--") && !Token::Match(tok->astParent(), "=|>>") &&
             !tok->astParent()->isUnaryOp("&"))
             return ExprUsage::Used;
-        if (Token::simpleMatch(tok->astParent(), "=") && astIsRHS(tok))
+        if (Token::simpleMatch(tok->astParent(), "=") && astIsRHS(tok)) {
+            if (tok->astParent()->astOperand1() && tok->astParent()->astOperand1()->variable() && tok->astParent()->astOperand1()->variable()->isReference())
+                return ExprUsage::NotUsed;
```

Comment:
It should still be `Used` even when its assigned to a reference. We will missing diagnosing cases like this:

```cpp
void f(int& x) {
    int i;
    x = i;
}
```

Reply:
See https://github.com/danmar/cppcheck/pull/5315

Reply:
can you add this test case to #5315?